### PR TITLE
feat: Add Continuous pre-training example for SageMaker hyperpod

### PR DESCRIPTION
**Pull request by Captainia**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: docs/source/training_tutorials/amazon_eks/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Set pip repository pointing to the Neuron repository
+RUN python -m pip config set global.extra-index-url https://pip.repos.neuron.amazonaws.com
+
+# Install wget, awscli
+RUN python -m pip install wget
+RUN python -m pip install awscli
+
+# Update Neuron Compiler and Framework
+RUN python -m pip install --upgrade neuronx-cc==2.* torch-neuronx==2.1.* torchvision
+RUN python -m pip install --upgrade neuronx-distributed neuronx-distributed-training
```
[the remark this answers is not in the document]
Comment (Captainia) [reply docs/source/training_tutorials/amazon_eks/Dockerfile:27]: Thank you both! Yes, this is not needed, as described in the description, this Dockerfile will need to be replaced with the new version of `huggingface-pytorch-training-neuronx` pre-built image. Once that is done, we do not need to upgrade these dependencies anymore. It is only needed here as we are depending on an old version of the image.

File: docs/source/training_tutorials/amazon_eks/llama3_train.yaml-template [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                - --dataset_config_name=${DATASET_CONFIG_NAME}
+                - --streaming=False
+                - --cache_dir=${TOKENIZED_DATA_PATH}
+                - --num_train_epochs=1
+                - --do_train
+                - --learning_rate=1e-4
+                - --max_steps=${MAX_STEPS}
+                - --per_device_train_batch_size=${BATCH_SIZE}
+                - --per_device_eval_batch_size=4
+                - --gradient_accumulation_steps=1
```
[the remark this answers is not in the document]
Comment (Captainia) [reply docs/source/training_tutorials/amazon_eks/llama3_train.yaml-template:154]: Yes, the eval batch size is not needed, I removed it from the argument. Regarding gradient accumulation steps, we tested both 1 and 8 so I updated it to 8, thanks for pointing this out.

File: docs/source/training_tutorials/pretraining_hyperpod_llm.mdx [hunk 3]
```diff
@@ -0,0 +1,194 @@
+<!---
+Copyright 2024 The HuggingFace Team. All rights reserved.
```
[the remark this answers is not in the document]
Comment (Captainia) [reply docs/source/training_tutorials/pretraining_hyperpod_llm.mdx:2]: Thank you!

File: docs/source/training_tutorials/amazon_eks/Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (tengomucho) [reply docs/source/training_tutorials/amazon_eks/Dockerfile]: I do not understand what is the bit the makes you require the `neuronx-distributed-training`. Can you please provide a link to it? Otherwise remove the dependency.

[the remark this answers is not in the document]
Comment (Captainia) [reply docs/source/training_tutorials/amazon_eks/Dockerfile]: Yes, you are right, we were including the upgrade for the older version, removed `neuronx-distributed-training` from the Docker file. Thanks!

**Review by tengomucho — APPROVED** [review]
LGTM thanks!

**Review by michaelbenayoun — APPROVED** [review]

**Comment by michaelbenayoun** [comment]
Merging but this should be updated when we release a training DLC.

**Comment by Captainia** [comment]
> Merging but this should be updated when we release a training DLC.

Thank you, yes absolutely. Once the DLC is available I will raise a separate PR to update.
